flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveGenericUDAF.java

```diff
@@ -105,6 +108,8 @@ private void init() throws HiveException {
                         GenericUDAFEvaluator.Mode.FINAL,
                         new ObjectInspector[] {partialResultObjectInspector});
 
+        isArgsSingleArray = HiveFunctionUtil.isSingleBoxedArray(argTypes);
```

Comment:
Array<Double>\Array<String>  ...  throws  exception  here

Reply:
Why throws exception here?

Reply:
for (DataType dataType : argTypes) {
            if (HiveFunctionUtil.isPrimitiveArray(dataType)) {
                throw new FlinkHiveUDFException(
                        "Flink doesn't support primitive array for Hive functions yet.");
            }
        }

Array<Double> is  PrimitiveArray , throws exception

Reply:
isArgsSingleArray = HiveFunctionUtil.isSingleBoxedArray(argTypes); 

Do you know why primiteive array is not supported?

if argsType is Array  then need the code `inputs = new Object[] {inputs}`;
if argsType is not Array then don't need the code.

Reply:
IIUC, the proble can be resolve in the changes of https://github.com/apache/flink/pull/19423/files#diff-fb0d46645c48e1c7b406addf9b31274625882afc69ad0cdf7700d2d67d699473R186

Reply:
I think we should determine whether the  argTypes  is Array , replace primitive Array type.`
If argsType is Array , inputs should be converted to an object[].
If argsType is not Array, we can keep origin code logic.

`HiveFunctionUtil.isSingleBoxedArray(argTypes);` throws exception when argTypes is Array and the elemet type is BOOLEAN | TINYINT | SMALLINT |  INTEGER | BIGINT | FLOAT | DOUBLE.

Reply:
Thanks, I see you.
Actuallu, I don't know why primiteive array is not supported. It's legacy code I have no knowledge.
According to my insight and test with Hive's qtest.  it should be supported. I think the validation can be dropped.
But currently, I would like to keep it and then see what will happen.

Reply:
Update: I prefer to drop such validation logic. The logic is weird and  the primitive array type should be supported from my test run with Hive's qtest. I think it may be legacy code.

Reply:
@luoyuxia , I am wrong.
Array(double) , the isNullable  of DoubleType is false，it will throw  exception.
Array(double) , the isNullable  of DoubleType is true，it is normal.

Reply:
If it is reasonable, can I go to the next step to support ?
`DataType[] argTypes = new DataType[]{DataTypes.ARRAY(DataTypes.INT().notNull())};`

https://issues.apache.org/jira/browse/FLINK-27238

Reply:
Yes, I think we has supported it. So I drop the check logic in this pr.